### ISSUE TITLE
Updating DotLiquid library version dependency

### ISSUE THREAD

**Issue by robin-parker** [Azure/logicapps#308]
Hi,
is it possible to specify a newer/explicit version of a nuget dependency for a V2 LogicApp?

I'm using the AzureFunctionApp task in an Azure DevOps Pipeline to deploy as a zip-archive, I cannot find documentation on how to specify explicit dependency versions.

https://www.nuget.org/packages/Microsoft.Azure.Workflows.WebJobs.Extension/1.1.2 shows a dependency on DotLiquid 2.0.361.
I need to use the latest DotLiquid version [2.1.450](https://www.nuget.org/packages/DotLiquid/2.1.450) as it has additional liquid filters and bug fixes.

Thanks

**Comment by shailesh-agre** [comment]
@robin-parker Product group is looking in to this request. Though there is no ETA right now, this will get implemented in near future.

**Comment by robin-parker** [comment]
Thanks @shailesh-agre, [this link](https://www.fuget.org/packages/DotLiquid/2.2.508/lib/net45/diff/2.0.361/) provides a summary of DotLiquid differences between 2.0.361 (as available in LogicApps) and the latest version of DotLiquid.

**Comment by daviburg** [comment]
> is it possible to specify a newer/explicit version of a nuget dependency for a V2 LogicApp?
> 
> I'm using the AzureFunctionApp task in an Azure DevOps Pipeline to deploy as a zip-archive, I cannot find documentation on how to specify explicit dependency versions.

That would be a generic ask for all NuGet dependencies, not just DotLiquid. Probably best to be on its own issue thread, separate from specific DotLiquid ask.

> I need to use the latest DotLiquid version [2.1.450](https://www.nuget.org/packages/DotLiquid/2.1.450) as it has additional liquid filters and bug fixes.

The team does need to update the specific DotLiquid dependency version but when it does this update it also need to enable two new features:

1. Specify the DotLiquid backward compatibility version target, per https://github.com/dotliquid/dotliquid/wiki/DotLiquid-Syntax-Compatibility
2. Specify additional filters to be loaded (such as the new Liquid filters) (@robin-parker there might not be any DotLiquid wiki explaining how to do that yet)

These are required because either has technically breaking changes from previous versions of DotLiquid.

**Comment by robin-parker** [comment]
> 2. Specify additional filters to be loaded (such as the new Liquid filters) (@robin-parker there might not be any DotLiquid wiki explaining how to do that yet)

@daviburg, I have made wiki changes to describe additional filters, they are ready to push after #444 is merged.

**Comment by robin-parker** [comment]
Hi @shailesh-agre,
is there any ETA on when the DotLiquid version could be updated for LogicApps Standard?

The existing dependency (dotliquid/2.0.361) is out of date as there have been numerous functional and performance changes since that was released in Jun'20.

Thanks,
Robin

**Comment by steftux** [comment]
Even the latest webjobs version (https://www.nuget.org/packages/Microsoft.Azure.Workflows.WebJobs.Extension/1.62.30 )is showing [DotLiquid](https://www.nuget.org/packages/DotLiquid/) (>= 2.0.361) . When will this ever be updated to a later version? I also need a few features of an updated DotLiquid...
